src/index.js

```diff
@@ -62,134 +117,128 @@ class Masq {
             await this.signout()
           }
         } else if (e.newValue) {
-          await this.connectToMasq()
+          await this._loadSessionInfo()
+          debug('[masq._listenForLoginOrSignout] dispatch logged_in')
           this.eventTarget.dispatchEvent(new Event('logged_in'))
         }
       }
     })
   }
 
   _listenForFocus () {
-    document.addEventListener('focus', () => {
-      this.connectToMasq()
+    document.addEventListener('focus', async () => {
+      // try to reopen db
+      if (!this.isLoggedIn()) {
+        return
+      }
+      await this._openDb()
+      this.startReplication()
     })
   }
 
-  _createSwarm (hub) {
-    return swarm(hub, {
-      config: this.config.swarmConfig
-    })
-  }
+  async _loadSessionInfo () {
+    const auxLoginFromUserInfo = async (currentUserInfo) => {
+      const { userAppDbId, userAppDEK, userAppNonce } = JSON.parse(currentUserInfo)
+      this.userAppDbId = userAppDbId
+      this.userAppDEK = userAppDEK
+      this.importedUserAppDEK = await common.crypto.importKey(Buffer.from(userAppDEK, 'hex'))
+      this.userAppNonce = userAppNonce
+      this.setState('logged')
+      await this._openDb()
+      this.startReplication()
+    }
 
-  _reset () {
-    this.userAppRepSW = null
-    this.userAppRepHub = null
+    // If user info is stored in session storage, do not use localStorage
+    const currentUserInfo = window.sessionStorage.getItem(CURRENT_USER_INFO_STR)
+    if (currentUserInfo) {
+      await auxLoginFromUserInfo(currentUserInfo)
+      return
+    }
 
-    this._logIntoMasqErr = null
-    this._logIntoMasqDone = false
+    // if userId is is not in session storage, look for it in local storage
+    // and save in session storage
+    const localStorageCurrentUserInfo = window.localStorage.getItem(CURRENT_USER_INFO_STR)
+    if (localStorageCurrentUserInfo) {
+      await auxLoginFromUserInfo(localStorageCurrentUserInfo)
+      window.sessionStorage.setItem(CURRENT_USER_INFO_STR, localStorageCurrentUserInfo)
+    }
   }
 
-  isLoggedIn () {
-    // is logged in if the userAppDbId is known
-    // (either after register or after construction and load session info)
-    return !!this.userId
+  async _openDb () {
+    try {
+      this.userAppDb = common.utils.createPromisifiedHyperDB(this.userAppDbId)
+      await common.utils.dbReady(this.userAppDb)
+    } catch (e) {
+      if (this.isLoggedIn()) {
+        throw e
+      }
+    }
   }
 
-  isConnected () {
-    // is connected if the userAppDb has been created and initialized
-    return !!this.userAppDb
+  _createSwarm (hub) {
+    return swarm(hub, {
+      config: this.config.swarmConfig
+    })
   }
 
-  async connectToMasq () {
-    await this._loadSessionInfo()
-
-    if (!this.isLoggedIn()) {
-      await this._disconnect()
+  startReplication () {
+    debug('[masq.startReplication]')
```

Comment:
should we add a precondition like this ?
```js
if (this.state === 'replicating') return
```

Reply:
For now executing start repliation when already replicating restarts replication